test2code-admin/src/main/kotlin/com/epam/drill/plugins/test2code/Plugin.kt

```diff
@@ -363,16 +363,20 @@ class Plugin(
             .also { logPoolStats() }
 
         is CoverDataPart -> {
-            message.data.groupBy { it.sessionId }
+            message.data
+                .groupBy { it.sessionId }
                 .forEach { (probeSessionId, data) ->
-                    val sessionId = probeSessionId ?: message.sessionId ?: GLOBAL_SESSION_ID
-                    activeScope.activeSessions[sessionId]?.let {
+                    var sessionId = probeSessionId ?: message.sessionId ?: GLOBAL_SESSION_ID
+                    if (activeScope.activeSessions[sessionId] == null){
+                        sessionId = GLOBAL_SESSION_ID
```

Comment:
Code is a bit contrived, since `sessionId` might already contain `GLOBAL_SESSION_ID` (so re-assigning it won't help at all)
Lets think of a better solution

Reply:
let`s impl new logic at new release